### ISSUE TITLE
Language Server error when importing SDL2 headers

### ISSUE THREAD

**Issue by SimantoR** [microsoft/vscode-cpptools#6167]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Windows 10 (Windows NT x64 10.0.19041)
- VS Code Version: 1.49.1
- C/C++ Extension Version: 1.0.1-insiders
- Other extensions you installed (and if the issue persists after disabling them): twxs.cmake (0.0.17) and ms-vscode.cmake-tools (1.4.2)
- Does this issue involve using SSH remote to run the extension on a remote machine?: No

The bug occurs when importing header files from selected projects (SDL2 headers and xtensor headers so far). When ever I try to include <SDL2/SDL.h>, my intellisense goes to simple C++ types and nothing more. None of the functions come up when pressing CTRL+Space. Everything works grand when I remove those headers again. Xtensor one was few weeks back and it wasn't important enough to dig into. If I write the function exactly and mouse-hover, information for that function appears correctly though.

**Expected behavior**
Intellisense for `SDL_Init(...)` showing exact spelling and parameter arguments

<details>
  <summary><strong>Logs</strong></summary>

```
Quick info operation failed: FE: 'Compiler exited with error - No IL available'
IntelliSense Engine = Default.
Enhanced Colorization is enabled.
Error squiggles are enabled if all header dependencies are resolved.
Autocomplete is enabled.
cpptools/didChangeCppProperties
$/setTraceNotification
cpptools/getCodeActions: file:///c%3A/Users/mdsim/Documents/SurrealEngine/src/main.cpp (id: 125)
cpptools/activeDocumentChange: file:///c%3A/Users/mdsim/Documents/SurrealEngine/src/main.cpp
cpptools/textEditorSelectionChange
cpptools/textEditorSelectionChange
cpptools/getFoldingRanges: file:///c%3A/Users/mdsim/Documents/SurrealEngine/src/main.cpp (id: 126)
cpptools/getCodeActions: file:///c%3A/Users/mdsim/Documents/SurrealEngine/src/main.cpp (id: 127)
idle loop: reparsing the active document
Checking for syntax errors: file:///c%3A/Users/mdsim/Documents/SurrealEngine/src/main.cpp
Queueing IntelliSense update for files in translation unit of: C:\USERS\MDSIM\DOCUMENTS\SURREALENGINE\SRC\MAIN.CPP
cpptools/finishUpdateSquiggles
Error squiggle count: 0
Update IntelliSense time (sec): 0.805
cpptools/getSemanticTokens: file:///c%3A/Users/mdsim/Documents/SurrealEngine/src/main.cpp (id: 128)
textDocument/hover: file:///c%3A/Users/mdsim/Documents/SurrealEngine/src/main.cpp (id: 129)
Quick info operation failed: FE: 'Compiler exited with error - No IL available'
using Tag Parser for quick info
cpptools/activeDocumentChange: file:///c%3A/Users/mdsim/Documents/SurrealEngine/src/main.cpp
cpptools/textEditorSelectionChange
textDocument/definition: file:///c%3A/Users/mdsim/Documents/SurrealEngine/src/main.cpp (id: 130)
textDocument/completion: file:///c%3A/Users/mdsim/Documents/SurrealEngine/src/main.cpp (id: 131)
auto_complete::handle_completion: file:///c%3A/Users/mdsim/Documents/SurrealEngine/src/main.cpp (155:13)
Offering completion
idle loop: reparsing the active document
Checking for syntax errors: file:///c%3A/Users/mdsim/Documents/SurrealEngine/src/main.cpp
Queueing IntelliSense update for files in translation unit of: C:\USERS\MDSIM\DOCUMENTS\SURREALENGINE\SRC\MAIN.CPP
cpptools/finishUpdateSquiggles
Error squiggle count: 0
Update IntelliSense time (sec): 0.812
cpptools/getSemanticTokens: file:///c%3A/Users/mdsim/Documents/SurrealEngine/src/main.cpp (id: 132)

```
</details>

**Screenshots**

![image](https://user-images.githubusercontent.com/34289663/93630062-ed190400-f9c3-11ea-97b5-5281f189bd50.png)
![image](https://user-images.githubusercontent.com/34289663/93630014-d377bc80-f9c3-11ea-9be5-166a1e5ed4f1.png)

**Additional context**

**Comment by charlesmacd** [comment]
I was just looking into this same problem. My setup is similar to yours, and I'm using MinGW for a compiler.
For me, this problem comes from the inclusion of `<intrin.h>` as follows:

* SDL2.h includes SDL_cpuinfo.h
* SDL_cpuinfo.h includes intrin.h on line 58:

```
#elif defined(__MINGW64_VERSION_MAJOR)
#include <intrin.h>
#else
```
Just commenting out the `#include <intrin.h>` in SDL_cpuinfo.h re-enables Intellisense for the entire project.

If I make a dummy program that only includes intrin.h and not any part of SDL2 it also breaks intellisense:

```
#include <stdio.h>
#include <intrin.h>
int main (int argc, char *argv[]) {
    printf() // if intrin.h is included, intellisense won't display anything, otherwise it will.
    return 0;
}
```
So in my case `intrin.h` is the problem, not really SDL2. I don't know what it is about that include file that trips up intellisense.
Are you affected in the same way?

**Comment by SimantoR** [comment]
I confirm on my end, the same thing @charlesmacd mentioned:
![image](https://user-images.githubusercontent.com/34289663/93672064-860e5480-fa82-11ea-85ce-e5b867b852c6.png)
![image](https://user-images.githubusercontent.com/34289663/93672075-a0483280-fa82-11ea-8195-63d822eab01e.png)

**Comment by sean-mcmanus** [comment]
This is most likely a duplicate of #6114 (fix available with 1.0.1-insiders) or #6147 (fix should be available with 1.0.1 later today).

**Comment by sean-mcmanus** [comment]
Can you update to 1.0.1 to see if this is fixed?

**Comment by charlesmacd** [comment]
I can confirm 1.0.1 fixed it so it was possibly related to #6147. It was not resolved with 1.0.1-insiders so probably was unrelated to #6114. 

Thanks for your help with this Sean!
